sanity/keylime-secure_mount/test.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+# vim: dict+=/usr/share/beakerlib/dictionary.vim cpt=.,w,b,u,t,i,k
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+SECURE_DIR="/var/lib/keylime/secure"
+KEYLIME_UNIT_FILE="/usr/lib/systemd/system/keylime_agent.service"
+
+rlJournalStart
+
+    rlPhaseStartSetup "Do the keylime setup"
+        rlRun 'rlImport "./test-helpers"' || rlDie "cannot import keylime-tests/test-helpers library"
+        rlAssertRpm keylime
+
+        # update /etc/keylime.conf
+        limeBackupConfig
+        rlRun "limeUpdateConf logger_root level DEBUG"
+        rlRun "limeUpdateConf logger_keylime level DEBUG"
+        rlRun "limeUpdateConf handler_consoleHandler level DEBUG"
+        # tenant
+        rlRun "limeUpdateConf tenant require_ek_cert False"
+        # if TPM emulator is present
+        if limeTPMEmulated; then
+            # start tpm emulator
+            rlRun "limeStartTPMEmulator"
+            rlRun "limeWaitForTPMEmulator"
+            rlRun "limeCondStartAbrmd"
+            # start ima emulator
+            rlRun "limeInstallIMAConfig"
+            rlRun "limeStartIMAEmulator"
+        fi
+        sleep 5
+        rlRun "limeStartVerifier"
+        rlRun "limeWaitForVerifier"
+        rlRun "limeStartRegistrar"
+        rlRun "limeWaitForRegistrar"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Agent create dir after removing and mount it"
+        rlRun "limeStartAgent"
+        rlRun "limeStopAgent"
+        rlRun "umount $SECURE_DIR"
+        rlRun "rm -rf $SECURE_DIR"
+        rlRun "cp $KEYLIME_UNIT_FILE $KEYLIME_UNIT_FILE-backup"
+        rlRun "sed -i '5,6d' $KEYLIME_UNIT_FILE"
+        rlRun "systemctl daemon-reload"
+        rlRun "limeStartAgent"
+        sleep 3
+        rlAssertGrep "INFO  keylime_agent::secure_mount > Directory \"$SECURE_DIR\" created" $(limeAgentLogfile)
+        rlRun "cat /proc/mounts | grep $SECURE_DIR"
+        rlRun "umount $SECURE_DIR"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Manual mount dir with wrong fs, agent fail"
+        rlRun "limeStopAgent"
+        rlRun "rm -rf $SECURE_DIR"
+        rlRun "mkdir -p $SECURE_DIR"
+        rlRun "mount -t ramfs -o size=1m,mode=0700 ramfs $SECURE_DIR"
+        rlRun "limeStartAgent"
```

Comment:
@kkaarreell Yeah agent failed with non-zer exit code :
```
[root@ci-vm-10-0-138-130 keylime-secure_mount]# systemctl status keylime_agent
● keylime_agent.service - The Keylime compute agent
     Loaded: loaded (/usr/lib/systemd/system/keylime_agent.service; disabled; preset: disabled)
    Drop-In: /etc/systemd/system/keylime_agent.service.d
             └─10-tcti.conf
             /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
             /etc/systemd/system/keylime_agent.service.d
             └─20-rust_log_trace.conf
     Active: activating (auto-restart) (Result: exit-code) since Thu 2023-06-01 03:09:53 EDT; 40s ago
    Process: 24053 ExecStart=/usr/bin/keylime_agent (code=exited, status=1/FAILURE)
   Main PID: 24053 (code=exited, status=1/FAILURE)
        CPU: 7ms
```